presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveUtil.java

```diff
@@ -308,13 +309,15 @@ private static void configureCompressionCodecs(JobConf jobConf)
 
             Class<? extends InputFormat<?, ?>> inputFormatClass = getInputFormatClass(jobConf, inputFormatName);
             if (symlinkTarget && inputFormatClass == SymlinkTextInputFormat.class) {
-                // Symlink targets are assumed to be TEXTFILE unless serde indicates otherwise.
-                inputFormatClass = TextInputFormat.class;
-                if (isDeserializerClass(schema, AvroSerDe.class)) {
-                    inputFormatClass = AvroContainerInputFormat.class;
+                String serDe = getDeserializerClassName(schema);
+                for (HiveStorageFormat format : EnumSet.allOf(HiveStorageFormat.class)) {
```

Comment:
It would be nice to have a test for this (e.g symlink that points to ORC file for example).

Reply:
I'll take care of this in a separate PR